kratos/processes/set_initial_state_process.h

```diff
@@ -0,0 +1,190 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics 
+//
+//  License:		 BSD License 
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Alejandro Cornejo
+//                    
+//
+
+#if !defined(KRATOS_SET_INITIAL_STATE_H_INCLUDED )
+#define  KRATOS_SET_INITIAL_STATE_H_INCLUDED
+
+// System includes
+
+// External includes
+
+// Project includes
+
+#include "processes/process.h"
+#include "includes/element.h"
+#include "includes/model_part.h"
+#include "includes/initial_state.h"
+#include "includes/mat_variables.h"
+
+namespace Kratos
+{
+
+///@name Kratos Classes
+///@{
+
+/// The SetInitialStateProcess.
+/** This Operation is a derived class from the process.h
+ *  
+*/
+template<SizeType TDim>
+class SetInitialStateProcess : public Process
+{
+public:
+    ///@name Type Definitions
+    ///@{
+
+    /// Pointer definition of Process
+    KRATOS_CLASS_POINTER_DEFINITION(SetInitialStateProcess);
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /// Default constructor.
+    SetInitialStateProcess(ModelPart& rModelPart) :
+        mrModelPart(model_part)
+    {
+        const SizeType voigt_size = (TDim == 3) ? 6 : 3;
+
+        mInitialStrain.resize(voigt_size, false);
+        mInitialStress.resize(voigt_size, false);
+        mInitialF.resize(TDim, TDim, false);
+
+        noalias(mInitialStrain) = ZeroVector(voigt_size);
+        noalias(mInitialStress) = ZeroVector(voigt_size);
+        noalias(mInitialF)      = ZeroMatrix(TDim, TDim);
+    }
+
+    /// Full constructor.
+    SetInitialStateProcess(ModelPart& rModelPart,
+                           const Vector& rInitialStrain,
+                           const Vector& rInitialStress,
+                           const Matrix& rInitialF) :
+        mrModelPart(model_part), mInitialStrain(rInitialStrain), 
+        mInitialStress(rInitialStress), mInitialF(rInitialF)
+    {
+    }
+
+    /// Destructor.
+    ~SetInitialStateProcess() override {}
+
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+    /// This operator is provided to call the process as a function and simply calls the Execute method.
+    void operator()()
+    {
+        Execute();
+    }
+
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+
+    /// this function will be executed at every time step BEFORE performing the solve phase
+    void ExecuteInitializeSolutionStep() override
+    {
+        const auto it_elem_begin = mrModelPart.ElementsBegin();
+        const auto& integration_points = it_elem_begin->GetGeometry().IntegrationPoints(it_elem_begin->GetIntegrationMethod());
+
+        #pragma omp parallel for
```

Comment:
please use parallel utilities

Reply:
do you have an example? :)

Reply:
Search for block_for_each, there are quite a few usages now